stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -781,6 +781,89 @@ void AsyncTask::dropInitialTaskExecutorPreferenceRecord() {
   swift_unknownObjectRelease(executorIdentityToRelease);
 }
 
+/******************************************************************************/
+/************************** TASK NAMING ***************************************/
+/******************************************************************************/
+
+void AsyncTask::pushInitialTaskName(const char* _taskName) {
+  assert(_taskName && "Task name must not be null!");
+  assert(hasInitialTaskNameRecord() && "Attempted pushing name but task has no initial task name flag!");
+
+  void *allocation = _swift_task_alloc_specific(
+      this, sizeof(class TaskNameStatusRecord));
+
+  // TODO: Copy the string maybe into the same allocation at an offset or retain the swift string?
```

Comment:
Retaining the Swift string makes sense to me. It avoids an extra copy when the string is dynamic, and avoids any copies when it's constant. I'm not sure how inconvenient it would be to manipulate a String from the C++ side of things.

Reply:
Yeah, so sadly it's quite painful to make this work... we have to manually inspect flags in the String's flags and then copy or not copy accordingly... I'll try to prepare our ABI so we COULD do this but I will probably not do so in the first impl.

Reply:
Is our LTO good enough to have the stdlib expose some minimal calls for us to use here, and have it optimize to something reasonable? (Or maybe this isn't performance sensitive enough for that to matter anyway?) We'd really want to reuse the existing `String` code and treat it as an opaque value here.